README.md

```diff
@@ -34,17 +55,82 @@ For example, to show the working tree status of the repository:
 
     $ ambit git status
 
+### Environment variables
+
+Optionally, 3 environment variables can be used to set custom paths.
+If a variable is not set, it will take up its default value as outlined:
+
+| Environment Variable | Description                            | Default                              |
+| -------------------- | -------------------------------------- | ------------------------------------ |
+| AMBIT_HOME_PATH      | Starting path of symlink destinations. | Home directory                       |
+| AMBIT_CONFIG_PATH    | Path to configuration file.            | `${HOME}/.config/ambit/config.ambit` |
+| AMBIT_REPO_PATH      | Path to dotfile repository directory.  | `${HOME}/.config/ambit/repo`         |
+
 ## Configuration
 
-`ambit` will search for a configuration file located at `${HOME}/.config/ambit/config.ambit`.
+The purpose of the configuration file is to set the paths of the symlinks.
+A symlink is defined with two parts: an existing file relative to `AMBIT_REPO_PATH`, and its destination relative to the system's home directory.
 
-## Installation
+### Configuration syntax examples
 
-Build the `ambit` binary from source:
+#### Basic match
 
-    $ git clone git@github.com:plamorg/ambit.git
-    $ cd ambit
-    $ cargo install --path .
+Symlink `${HOME}/host.txt -> ${AMBIT_REPO_PATH}/a/repo.txt`:
+
+    a/repo.txt => host.txt;
+
+#### Implicit match
+
+If no `=>` operator is provided, it is assumed that the path given is both the `HOME` and `REPO` path.
+
+Symlink `${HOME}/.config/ambit/config.ambit -> ${AMBIT_REPO_PATH}/.config/ambit/config.ambit`:
+
+    .config/ambit/config.ambit;
+
+#### Variant Expression
+
+Variant expressions, denoted with brackets (`[]`), are a shorthand way to set multiple paths.
+
+Symlink:
```

Comment:
This should be changed to "This code can be shortened by using a variant expression..." or similar.